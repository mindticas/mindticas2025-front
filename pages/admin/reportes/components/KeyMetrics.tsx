import { Flex } from '@chakra-ui/react';
import StatsCard from './StatsCard';
import { CheckCircle, DollarSign, XCircle, Scissors } from 'lucide-react';
import { StatisticsDataResponse } from '@/interfaces/statistics/StatisticsDataResponse';

interface KeyMetricsProps {
    statistics: StatisticsDataResponse | null;
}

export default function KeyMetrics({ statistics }: KeyMetricsProps) {
    const statsItems = [
        {
            title: 'Ganancias',
            value: statistics?.totalEarnings ?? 0,
            icon: DollarSign,
            color: 'green.500',
        },
        {
            title: 'Servicios',
            value: statistics?.totalServices ?? 0,
            icon: Scissors,
            color: 'blue.500',
        },
        {
            title: 'Citas completadas',
            value: statistics?.totalCompletedAppointments ?? 0,
            icon: CheckCircle,
            color: 'green.500',
        },
        {
            title: 'Citas canceladas',
            value: statistics?.totalCanceledAppointments ?? 0,
            icon: XCircle,
            color: 'red.500',
        },
    ];

    return (
        <>
            <Flex
                gap={['4', '6', '10']}
                mx={['4', '6', '8', '12']}
                p='4'
                mt={['6', '14']}
                direction={['column', 'row', 'row', 'row']}
            >
                {statsItems.map((item, index) => (
                    <StatsCard
                        key={index}
                        title={item.title}
                        value={item.value}
                        icon={item.icon}
                        color={item.color}
                    />
                ))}
            </Flex>
        </>
    );
}
