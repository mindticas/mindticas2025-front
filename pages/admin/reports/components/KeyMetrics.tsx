import { Flex } from '@chakra-ui/react';
import StatsCard from './StatsCard';
import {
    CheckCircle,
    DollarSign,
    XCircle,
    Scissors,
    HandCoins,
    Store,
} from 'lucide-react';
import { StatisticsDataResponse } from '@/interfaces/statistics/StatisticsDataResponse';
interface KeyMetricsProps {
    statistics: StatisticsDataResponse | null;
    isLoading?: boolean;
}

export default function KeyMetrics({
    statistics,
    isLoading = false,
}: KeyMetricsProps) {
    const statsItems = [
        {
            title: 'Ganancias en servicios',
            value: statistics?.totalEarnings ?? 0,
            icon: DollarSign,
            color: 'green.500',
        },
        {
            title: 'Servicios',
            value: statistics?.totalServices ?? 0,
            icon: Scissors,
            color: 'blue.500',
        },
        {
            title: 'Citas completadas',
            value: statistics?.totalCompletedAppointments ?? 0,
            icon: CheckCircle,
            color: 'green.500',
        },
        {
            title: 'Citas canceladas',
            value: statistics?.totalCanceledAppointments ?? 0,
            icon: XCircle,
            color: 'red.500',
        },
        {
            title: 'Ventas de productos',
            value: statistics?.totalSalesAmount ?? 0,
            icon: Store,
            color: 'green.500',
        },
        {
            title: 'Propinas',
            value: statistics?.totalTips ?? 0,
            icon: HandCoins,
            color: 'yellow.500',
        },
    ];

    return (
        <>
            <Flex
                gap={['4', '6', '10']}
                mx={['4', '6', '8', '12']}
                p='4'
                mt={['6', '14']}
                direction={['column', 'column', 'column', 'row']}
            >
                {statsItems.map((item, index) => (
                    <StatsCard
                        key={index}
                        title={item.title}
                        value={item.value}
                        icon={item.icon}
                        color={item.color}
                        isLoading={isLoading}
                    />
                ))}
            </Flex>
        </>
    );
}
