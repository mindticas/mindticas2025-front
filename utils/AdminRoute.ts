import Layout from '@/components/Layout';
import { Box } from '@chakra-ui/react';

export class AdminRoute {
    // Fixed route for admin
    private static route: string = '/admin';

    // Returns the Layout component if the pathname contains '/admin', otherwise returns box
    static getLayout(pathname: string) {
        return pathname.includes(this.route) ? Layout : Box;
    }
}
