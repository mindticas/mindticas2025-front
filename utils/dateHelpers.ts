// Gets the name of the day according to the full calendar format
export const getDayName = (dateStr: string): string => {
    const [year, month, day] = dateStr.split('-').map(Number)
    const date = new Date(year, month -1, day)
    return date.toLocaleDateString('es-ES', { weekday: 'long'})
}
// Get the milliseconds in a day
export function addDays(date: Date, days: number): Date {
    const milisecondsPerDay = 24 * 60 * 60 * 1000; 
    return new Date(date.getTime() + days * milisecondsPerDay)
}