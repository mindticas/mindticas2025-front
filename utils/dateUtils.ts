import { DateTime } from "luxon";

interface DateTimeParts {
    date: string;
    time: string;
}

// * @returns Object with (YYYY-MM-DD) y time (HH:MM)

export const splitDateTimeFromISO = (
    isoString: string | null | undefined,
): DateTimeParts => {
    if (!isoString) return { date: '', time: '' };

    const dateObj = new Date(isoString);

    if (isNaN(dateObj.getTime())) {
        return { date: '', time: '' };
    }

    // format date (YYYY-MM-DD)
    const date = DateTime.fromJSDate(dateObj)
       .setZone('local')
        .toFormat('yyyy-MM-dd');

    // format time (HH:MM)
    const time = dateObj
        .toLocaleTimeString('es-ES', {
            hour: '2-digit',
            minute: '2-digit',
            hour12: false,
        })
        .replace(/\./g, '');

    return { date, time };
};
