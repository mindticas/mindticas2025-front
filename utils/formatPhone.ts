export default function formatPhone(phone: string): string {
    // Remove everything that is not digits
    const digits = phone.replace(/\D/g, '');
    // Applies format only if it has 10 digits
    if (digits.length === 10) {
        return `${digits.slice(0, 3)}-${digits.slice(3, 6)}-${digits.slice(6)}`;
    }
    return phone; // Returns the original if it doesn't match
}
