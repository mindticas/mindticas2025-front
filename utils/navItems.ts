import { BarChart, Calendar, Scissors, UserCog, Users } from 'lucide-react';

export const navItems = [
    { name: 'Citas', href: '/admin/', icon: Calendar },
    { name: 'Clientes', href: '/admin/clients', icon: Users },
    { name: 'Empleados', href: '/admin/employees', icon: UserCog },
    { name: 'Tratamientos', href: '/admin/treatments', icon: Scissors },
    { name: 'Reportes', href: '/admin/reports', icon: BarChart },
];
