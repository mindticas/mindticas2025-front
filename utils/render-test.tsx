import { Provider } from '@/components/ui/provider';
import { BookingProvider } from '@/context/BookingContext';
import { render as rtlRender } from '@testing-library/react';

export function render(ui: React.ReactNode) {
    return rtlRender(<>{ui}</>, {
        wrapper: (props: React.PropsWithChildren) => (
            <Provider>
                <BookingProvider>{props.children}</BookingProvider>
            </Provider>
        ),
    });
}
