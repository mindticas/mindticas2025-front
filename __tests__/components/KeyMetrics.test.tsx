import KeyMetrics from '@/pages/admin/reports/components/KeyMetrics';
import { render } from '@/utils/render-test';

describe('KeyMetrics Snapshot', () => {
    const mockStatistics = {
        totalEarnings: 1234.56,
        totalServices: 78,
        totalCompletedAppointments: 65,
        totalCanceledAppointments: 13,
    };

    it('renders correctly with statistics data', () => {
        const { asFragment } = render(
            <KeyMetrics statistics={mockStatistics} isLoading={false} />,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly while loading', () => {
        const { asFragment } = render(
            <KeyMetrics statistics={null} isLoading={true} />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
