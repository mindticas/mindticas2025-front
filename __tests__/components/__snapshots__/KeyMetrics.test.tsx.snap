// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyMetrics Snapshot renders correctly while loading 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-13vdia"
  >
    <div
      class="css-1b8npii"
    >
      <div
        class="css-1y96gi1"
      >
        <div
          class="css-1cteixr"
        >
          <svg
            class="lucide lucide-dollar-sign"
            fill="none"
            height="24"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="22"
            />
            <path
              d="M17 5H9.5a3.5 3.5 0 0 0 0 7h5a3.5 3.5 0 0 1 0 7H6"
            />
          </svg>
        </div>
      </div>
      <p
        class="css-ycbeac"
      >
        Ganancias
      </p>
      <div
        class="css-1vhhqj7"
      >
        <span
          class="chakra-spinner css-15lzuyp"
        />
      </div>
    </div>
    <div
      class="css-ctvivy"
    >
      <div
        class="css-1y96gi1"
      >
        <div
          class="css-1lsb2en"
        >
          <svg
            class="lucide lucide-scissors"
            fill="none"
            height="24"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="6"
              cy="6"
              r="3"
            />
            <path
              d="M8.12 8.12 12 12"
            />
            <path
              d="M20 4 8.12 15.88"
            />
            <circle
              cx="6"
              cy="18"
              r="3"
            />
            <path
              d="M14.8 14.8 20 20"
            />
          </svg>
        </div>
      </div>
      <p
        class="css-ycbeac"
      >
        Servicios
      </p>
      <div
        class="css-1vhhqj7"
      >
        <span
          class="chakra-spinner css-1uxiipu"
        />
      </div>
    </div>
    <div
      class="css-1b8npii"
    >
      <div
        class="css-1y96gi1"
      >
        <div
          class="css-1cteixr"
        >
          <svg
            class="lucide lucide-circle-check-big"
            fill="none"
            height="24"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21.801 10A10 10 0 1 1 17 3.335"
            />
            <path
              d="m9 11 3 3L22 4"
            />
          </svg>
        </div>
      </div>
      <p
        class="css-ycbeac"
      >
        Citas completadas
      </p>
      <div
        class="css-1vhhqj7"
      >
        <span
          class="chakra-spinner css-15lzuyp"
        />
      </div>
    </div>
    <div
      class="css-122d0pt"
    >
      <div
        class="css-1y96gi1"
      >
        <div
          class="css-1lgm3s8"
        >
          <svg
            class="lucide lucide-circle-x"
            fill="none"
            height="24"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <path
              d="m15 9-6 6"
            />
            <path
              d="m9 9 6 6"
            />
          </svg>
        </div>
      </div>
      <p
        class="css-ycbeac"
      >
        Citas canceladas
      </p>
      <div
        class="css-1vhhqj7"
      >
        <span
          class="chakra-spinner css-e3hgfy"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`KeyMetrics Snapshot renders correctly with statistics data 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-13vdia"
  >
    <div
      class="css-1b8npii"
    >
      <div
        class="css-1y96gi1"
      >
        <div
          class="css-1cteixr"
        >
          <svg
            class="lucide lucide-dollar-sign"
            fill="none"
            height="24"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="22"
            />
            <path
              d="M17 5H9.5a3.5 3.5 0 0 0 0 7h5a3.5 3.5 0 0 1 0 7H6"
            />
          </svg>
        </div>
      </div>
      <p
        class="css-ycbeac"
      >
        Ganancias
      </p>
      <div
        class="css-1vhhqj7"
      >
        <p
          class="css-158617b"
        >
          1234.56
        </p>
      </div>
    </div>
    <div
      class="css-ctvivy"
    >
      <div
        class="css-1y96gi1"
      >
        <div
          class="css-1lsb2en"
        >
          <svg
            class="lucide lucide-scissors"
            fill="none"
            height="24"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="6"
              cy="6"
              r="3"
            />
            <path
              d="M8.12 8.12 12 12"
            />
            <path
              d="M20 4 8.12 15.88"
            />
            <circle
              cx="6"
              cy="18"
              r="3"
            />
            <path
              d="M14.8 14.8 20 20"
            />
          </svg>
        </div>
      </div>
      <p
        class="css-ycbeac"
      >
        Servicios
      </p>
      <div
        class="css-1vhhqj7"
      >
        <p
          class="css-158617b"
        >
          78
        </p>
      </div>
    </div>
    <div
      class="css-1b8npii"
    >
      <div
        class="css-1y96gi1"
      >
        <div
          class="css-1cteixr"
        >
          <svg
            class="lucide lucide-circle-check-big"
            fill="none"
            height="24"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21.801 10A10 10 0 1 1 17 3.335"
            />
            <path
              d="m9 11 3 3L22 4"
            />
          </svg>
        </div>
      </div>
      <p
        class="css-ycbeac"
      >
        Citas completadas
      </p>
      <div
        class="css-1vhhqj7"
      >
        <p
          class="css-158617b"
        >
          65
        </p>
      </div>
    </div>
    <div
      class="css-122d0pt"
    >
      <div
        class="css-1y96gi1"
      >
        <div
          class="css-1lgm3s8"
        >
          <svg
            class="lucide lucide-circle-x"
            fill="none"
            height="24"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <path
              d="m15 9-6 6"
            />
            <path
              d="m9 9 6 6"
            />
          </svg>
        </div>
      </div>
      <p
        class="css-ycbeac"
      >
        Citas canceladas
      </p>
      <div
        class="css-1vhhqj7"
      >
        <p
          class="css-158617b"
        >
          13
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
